src/acl_mem.cpp

```diff
@@ -405,6 +405,88 @@ int acl_bind_buffer_to_device(cl_device_id device, cl_mem mem) {
   return 1;
 }
 
+ACL_EXPORT
+// CL_API_ENTRY cl_int clEnqueueReadGlobalVariableINTEL() {
+CL_API_ENTRY cl_int clEnqueueReadGlobalVariableINTEL(
+    cl_command_queue command_queue,
+    cl_program program,
+    const char* name,
+    cl_bool blocking_write,
+    size_t size,
+    size_t offset,
+    void* ptr,
+    cl_uint num_events_in_wait_list,
+    const cl_event* event_wait_list,
+    cl_event* event,
+    const void* dev_global_ptr) {
+    
+  // TODO: get dev_global_ptr from autodiscovery instead later
+  // return 0;
+  return clEnqueueWriteGlobalVariableINTEL(
+    command_queue, program, name, blocking_write,
+    size, offset, ptr, num_events_in_wait_list,
+    event_wait_list, event, dev_global_ptr);
+}
+
+ACL_EXPORT
+CL_API_ENTRY cl_int clEnqueueWriteGlobalVariableINTEL(
+    cl_command_queue command_queue,
+    cl_program program,
+    const char* name,
+    cl_bool blocking_write,
+    size_t size,
+    size_t offset,
+    const void* ptr,
+    cl_uint num_events_in_wait_list,
+    const cl_event* event_wait_list,
+    cl_event* event,
+    const void* dev_global_ptr) {
+  cl_int status;
+  cl_kernel kernel = clCreateKernelIntelFPGA(program, name, &status);
+  if (status != CL_SUCCESS) {
+    return status;
+  }
+  // TODO: get dev_global_ptr from autodiscovery instead later
+  // Assuming below returns same thing as `clDeviceMemAllocINTEL`, where exactly is dev global ptr being read from?
+  // What format is the read dev global, is it a pointer just like the return value of `clDeviceMemAllocINTEL` or is it something else?
+  // void* dev_global_ptr = kernel->dev_bin->get_devdef().autodiscovery_def.?
```

Comment:
What exact format is being read from autodiscovery? Is it exactly the same as the return of `clDeviceMemAllocINTEL`?